### ISSUE TITLE
Kubespawner does not allow me to set API_URL

### ISSUE THREAD

**Issue by crauterb** [jupyterhub/kubespawner#359]
# Expected Behavior

I can define a value for the Environment Variable `JUPYTERHUB_API_URL` to specify the URL used to contact the API from within a spawned user pod.

# Actual Behavior

Despite me setting according values in my config:
```
c.KubeSpawner.environment = {
    'JUPYTERHUB_API_URL': 'service-name.namespace:8081/hub/api',
    'TEST_NAME': 'HERBERT'
}
```
the only the `TEST_NAME` is set in the Pod, the other value is - I am guessing - overwritten by the default from the SuperClass in `spawner.py`:
```
    Environment:
      JUPYTERHUB_API_URL:             http://127.0.0.1:8081/hub/api
      TEST_NAME:                      HERBERT
```

# Steps to Reproduce the Problem

1. Create a deployment using `jupyterhub/jupyterhub:1.0.0`
2. Add a `jupyterhub_config.py` with:
```
c.JupyterHub.spawner_class = 'kubespawner.KubeSpawner'
c.KubeSpawner.debug = True
c.Spawner.default_url = '/lab'
c.KubeSpawner.pod_name_template = 'jupyter-{username}'
c.KubeSpawner.cmd = 'jupyter-labhub'
c.KubeSpawner.image_pull_policy = 'Always'
c.KubeSpawner.image = 'jupyterhub/singleuser:1.0'
c.KubeSpawner.start_timeout = 60 * 5
c.KubeSpawner.environment = {
    'JUPYTERHUB_API_URL': 'ceb-jupyterhub.ceb-jupyter:8081/hub/api',
    'TEST_NAME': 'HERBERT'
}
```
3. Spawn user pod.

# Specifications
*Version*: JupyterHub 1.0.0, Kubespawner 0.10
*Platform*: Kubernetes 1.11

# Additional Notes

I am currently setting up a deployment of JupyterHub in Kubernetes. 
As I - initially - tried to get everything working with Istio integration, I chose not to use [the zero-to-jupyterhub-configuration](https://zero-to-jupyterhub.readthedocs.io/en/latest/).

However, I am still stuck at this specific point, Istio or no Istio: I can not find the correct way to define the API URL for the spawned pods.

Setting `c.ConfigurableHTTPProxy.api_url = '0.0.0.0:8001'` also did not help.

I do hope that there is no oversight on my part (-:
Any input is much appreciated.

**Comment by drew-simmons** [comment]
@crauterb were you able to find a way to set the kubernetes api?

**Comment by consideRatio** [comment]
`JUPYTERHUB_API_URL` is an environment variable automatically set by the JupyterHub Spawner baseclass that KubeSpawner inherits from. This is done [here in the code](https://github.com/jupyterhub/jupyterhub/blob/0317fd63fa1dac4d709b4f95738e74c5c408da6c/jupyterhub/spawner.py#L766).

In JupyterHub 1.2, currently unreleased, the spawners `.environment` configuration will get the highest priority though so an override of this will be possible. See [this commit to JupyterHub's Spawner base class](https://github.com/jupyterhub/jupyterhub/commit/df35268bfee2ae9eb6ab51ac29e9521b8ae8ed1f).

I think the current JUPYTERHUB_API_URL is set through the configuration of `c.JupyterHub.ip` and `c.JupyterHub.port`, but I'm not 100% sure.

---

I'm closing this as something resolved in JupyterHub 1.2 to be released, but I think this can be configured in JupyterHub as well, perhaps using the `.ip` and `.port` configuration or `hub_connect_url` which indirectly set the `.ip` `.port` I think.
